components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,388 @@
+        metricReporterInstance.incrementCount(MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL, new String[]{name,
+                SynapseConstants.PROXY_SERVICE_TYPE});
+    }
+
+    /**
+     * Increment the error request count received by an api.
+     *
+     * @param name             The metric name
+     * @param apiInvocationUrl api Invocation URL
+     */
+    private void incrementApiErrorCount(String name, String apiInvocationUrl) {
+        metricReporterInstance.incrementCount(MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL,  new String[]{name,
+                SynapseConstants.FAIL_SAFE_MODE_API, apiInvocationUrl});
+    }
+
+    /**
+     * Increment the error request count received by an inbound endpoint.
+     *
+     * @param name The metric name
+     */
+    private void incrementInboundEndpointErrorCount(String name) {
+        metricReporterInstance.incrementCount(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL,
+                new String[]{name, MetricConstants.INBOUND_ENDPOINT});
+    }
+
+    /**
+     * Get the api name.
+     *
+     * @param contextPath The api context path
+     * @param synCtx The Synapse Message Context
+     * @return String The api name
+     */
+    private String getApiName(String contextPath, MessageContext synCtx) {
+        String apiName = null;
+        for (API api : synCtx.getEnvironment().getSynapseConfiguration().getAPIs()) {
+            if (RESTUtils.matchApiPath(contextPath, api.getContext())) {
+                apiName = api.getName();
+                if (api.getVersionStrategy().getVersion() != null && !"".equals(api.getVersionStrategy().
+                        getVersion())) {
+                    apiName = apiName + ":v" + api.getVersionStrategy().getVersion();
+                }
+                synCtx.setProperty(RESTConstants.IS_PROMETHEUS_ENGAGED, true);
+                synCtx.setProperty(RESTConstants.PROCESSED_API, api);
+            }
+        }
+        return apiName;
+    }
+
+    /**
+     * Return the port the service was invoked.
+     *
+     * @param synCtx Synapse message context
+     * @return port the port used to invoke the service
+     */
+    private int getServiceInvokePort(MessageContext synCtx) {
+        if (null != ((Axis2MessageContext) synCtx).getAxis2MessageContext().
+                getProperty(NhttpConstants.SERVICE_PREFIX)) {
+            String servicePort = ((Axis2MessageContext) synCtx).getAxis2MessageContext().
+                    getProperty(NhttpConstants.SERVICE_PREFIX).toString();
+            servicePort = servicePort.substring((servicePort.lastIndexOf(':') + 1),
+                    servicePort.lastIndexOf(DELIMITER));
+            serviceInvokePort = Integer.parseInt(servicePort);
+        }
+        return serviceInvokePort;
+    }
+
+    /**
+     * Return the internal http api port.
+     *
+     * @return port internal http api port
+     */
+    public static int getInternalHttpApiPort() {
```

Comment:
This method does not have to be static anymore. WDYT?

Reply:
Fixed